evdns.c

```diff
@@ -671,21 +675,18 @@ nameserver_probe_failed(struct nameserver *const ns) {
 		return;
 	}
 
-#define MAX_PROBE_TIMEOUT 3600
-#define TIMEOUT_BACKOFF_FACTOR 3
-
 	memcpy(&timeout, &ns->base->global_nameserver_probe_initial_timeout,
 	    sizeof(struct timeval));
-	for (i=ns->failed_times; i > 0 && timeout.tv_sec < MAX_PROBE_TIMEOUT; --i) {
-		timeout.tv_sec *= TIMEOUT_BACKOFF_FACTOR;
-		timeout.tv_usec *= TIMEOUT_BACKOFF_FACTOR;
+	for (i=ns->failed_times; i > 0 && timeout.tv_sec < ns->base->ns_max_probe_timeout; --i) {
```

Comment:
Since you already touched this line, let's use more preferable syntax (over deprecated):

```suggestion
	for (i = ns->failed_times; i > 0 && timeout.tv_sec < ns->base->ns_max_probe_timeout; --i) {
```